### PR TITLE
Infinite scroll on work item list added

### PR DESCRIPTION
**Pull request by sanbornsen**
- Output events to update the list removed from quickadd.
- Output events to update the list removed from wi entry on delete.
- Using reference of the list object all over the places to update the dom.
- Added a new directive for handling infinite scroll event and page size.

### CODE REVIEW

File: src/app/work-item/work-item-list/work-item-list.component.ts [hunk 1]
```diff
@@ -48,31 +50,41 @@ export class WorkItemListComponent implements OnInit {
...
   }
 
   // model handlers
 
-  reloadWorkItems(): void {
+  initWiItems(event: any): void {
+    this.loadWorkItems(event.pageSize);
+  }
+
+  loadWorkItems(pageSize: number = 20): void {
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item-list/work-item-list.component.ts]: pageSize is defined as 10 in infinitescroll.directive. Why is it reset to 20 here?

File: src/app/work-item/work-item.service.ts [hunk 2]
```diff
@@ -28,18 +30,57 @@ export class WorkItemService {
...
+          }
+          this.workItems = response.json().data as WorkItem[];
+        }
         return this.workItems;
       })
       .catch(this.handleError);
   }
 
+  getMoreWorkItems(): Promise<any> {
+    console.log(this.nextLink);
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item.service.ts]: Remove the console.log statement

File: src/app/work-item/work-item.service.ts [hunk 3]
```diff
@@ -83,11 +124,16 @@ export class WorkItemService {
...
+      .then(() => {
+        let deletedItemIndex = this.workItems.findIndex((item) => item.id == workItem.id);
+        // removing deleted item from the local list
+        this.workItems.splice(deletedItemIndex, 1);
+      })
       .catch(this.handleError);
   }
 
   create(workItem: WorkItem): Promise<WorkItem> {
+    console.log(workItem);
```
Comment (nimishamukherjee) [inline src/app/work-item/work-item.service.ts]: Remove console.log statement

Comment (sanbornsen) [inline src/app/work-item/work-item.service.ts]: Yeah, this one can go off.

File: src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.ts [hunk 4]
```diff
@@ -213,7 +213,6 @@ export class WorkItemDetailComponent implements OnInit {
       .update(this.workItem)
       .then((workItem) => {
         this.workItem.version = workItem.version;
-        this.broadcaster.broadcast('updateWorkItem', workItem);
```
Comment (sanbornsen) [inline src/app/work-item/work-item-list/work-item-detail/work-item-detail.component.ts]: We are now updating values by reference not by events.

File: src/app/work-item/work-item-list/work-item-list.component.ts (hunk 1 above)
Comment (sanbornsen) [inline src/app/work-item/work-item-list/work-item-list.component.ts]: Also the pageSize 10 in the directive is just a default value.

**Pull request merged**